### PR TITLE
task3

### PR DESCRIPTION
**Pull request by DmitryChep**
I think that, as always, you can make it more elegant) But everything works like that.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,195 @@
+######################
```
Comment (AlexChGit) [inline .gitignore:1]: Please check previous Pull request about .gitignore file..

File: src/main/java/l2/Task3.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import l2.math.PrimeCheck;
+
+public class Task3 {
+    public static void main(String[] args) {
+        PrimeCheck primeCheck = new PrimeCheck();
+
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Please enter in some number");
+        int number = scanner.nextInt();
+        int num = number;
```
Comment (AlexChGit) [inline src/main/java/l2/Task3.java:13]: the same.. there is a way to make this two above lines better..

File: src/main/java/l2/Task3.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        if (num ==1){
```
Comment (AlexChGit) [inline src/main/java/l2/Task3.java:14]: add a Space after ==

File: src/main/java/l2/math/PrimeCheck.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package l2.math;
+
+public class PrimeCheck {
+
+    public boolean check(int num) {
+        boolean a = true, b = false ;
+        for (int i = 2; i < num ; i++){
```
Comment (AlexChGit) [inline src/main/java/l2/math/PrimeCheck.java:7]: add a Space before curly brace

**Comment by DmitryChep** [comment]
Ok thanks)

File: src/main/java/l2/math/PrimeCheck.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package l2.math;
+
+public class PrimeCheck {
+
+    public boolean check(int num) {
```
Comment (AlexChGit) [inline src/main/java/l2/math/PrimeCheck.java:5]: not fully correct logic of the method.. try to do better..

File: src/main/java/l2/Task3.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    public static void main(String[] args) {
+        PrimeCheck primeCheck = new PrimeCheck();
+
+        Scanner scanner = new Scanner(System.in);
+        System.out.print("Please enter in some number: ");
+        int num  = scanner.nextInt();
+        if (num == 1) {
+            System.out.println("This number is neither a prime number nor a composite number.");
+        } else {
+            if (primeCheck.check(num) == true) {
```
Comment (AlexChGit) [inline src/main/java/l2/Task3.java:16]: You can remove "== true" and you will get the same result. Check!

File: src/main/java/l2/Task3.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+import l2.math.PrimeCheck;
+
+public class Task3 {
+    public static void main(String[] args) {
+        PrimeCheck primeCheck = new PrimeCheck();
+
+        Scanner scanner = new Scanner(System.in);
+        System.out.print("Please enter in some number: ");
+        int num  = scanner.nextInt();
+        if (num == 1) {
```
Comment (AlexChGit) [inline src/main/java/l2/Task3.java:13]: Two lines above could be optimized. Try to do that.

File: src/main/java/l2/math/PrimeCheck.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+        boolean a = true, b = false ;
+        for (int i = 2; i < num ; i++) {
+            int r = num % i;
+            if (r != 0){
```
Comment (AlexChGit) [inline src/main/java/l2/math/PrimeCheck.java:9]: Add Space before curly brace

File: src/main/java/l2/math/PrimeCheck.java [hunk 9]
```diff
@@ -0,0 +1,16 @@
+package l2.math;
+
+public class PrimeCheck {
+
+    public boolean check(int num) {
+        boolean a = true, b = false ;
+        for (int i = 2; i < num ; i++) {
```
Comment (AlexChGit) [inline src/main/java/l2/math/PrimeCheck.java:7]: Remove Space after "num"

File: src/main/java/l2/math/PrimeCheck.java [hunk 10]
```diff
@@ -0,0 +1,16 @@
+package l2.math;
+
+public class PrimeCheck {
+
+    public boolean check(int num) {
+        boolean a = true, b = false ;
```
Comment (AlexChGit) [inline src/main/java/l2/math/PrimeCheck.java:6]: Remove Space after "false"

File: src/main/java/l2/Task3.java (hunk 7 above)
Comment (DmitryChep) [reply src/main/java/l2/Task3.java:13]: To be honest i try to understund but not succesfuly) Maybe i don't have a knowlege

Comment (DmitryChep) [reply src/main/java/l2/Task3.java:13]: You meant 12 and 13 lines? I am coorectly understund?

Comment (AlexChGit) [reply src/main/java/l2/Task3.java:13]: Yes, correctly. if (scanner.nextInt() == 1) {...}

File: src/main/java/l2/Task3.java [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+import l2.math.PrimeCheck;
+
+public class Task3 {
+    public static void main(String[] args) {
+        PrimeCheck primeCheck = new PrimeCheck();
+
+        Scanner scanner = new Scanner(System.in);
+        System.out.print("Please enter in some number: ");
+        int num = scanner.nextInt();
+        if(scanner.nextInt() == 1) {
```
Comment (AlexChGit) [inline src/main/java/l2/Task3.java:13]: Change to (num == 1) back, becouse you aslo use "num" below. But you need to know that you can use methods invokations (scanner.nextInt()) directly in structures (if, for, while etc.).

Comment (DmitryChep) [reply src/main/java/l2/Task3.java:13]: Got it! thanks.

**Pull request merged**